test/backoff.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const assert = require('assert');
+
+const Firestore = require('../src');
+const backoff = require('../src/backoff')(Firestore);
+
+const ExponentialBackoff = backoff.ExponentialBackoff;
+const setTimeoutHandler = backoff.setTimeoutHandler;
+
+const nop = () => {};
+
+describe('ExponentialBackoff', function() {
+  const observedDelays = [];
+
+  before(() => {
+    setTimeoutHandler((callback, timeout) => observedDelays.push(timeout));
+  });
+
+  beforeEach(() => {
+    observedDelays.length = 0;
+  });
+
+  after(() => {
+    setTimeoutHandler(setTimeout);
+  });
+
+  function assertDelayEquals(expected) {
+    assert.equal(observedDelays.shift(), expected);
+  }
+
+  function assertDelayBetween(low, high) {
+    let actual = observedDelays.shift();
+    assert.ok(actual >= low);
```

Comment:
FWIW (probably not much), I'm surprised we're not using a more powerful assertion framework (chai or whatever) which would let us do a proper expect(actual).toBeGreaterThanOrEqualTo(low) or whatever, with the advantage that you get better error messages on failure (rather than "expected false to be true" or whatever)

Reply:
Now that we are in our own repo, I'm sure that we can migrate to chai. At some point that is.